cmd/runtimetest/main.go

```diff
@@ -439,12 +454,15 @@ func testWriteAccess(path string) (writable bool, err error) {
 	if err != nil {
 		return false, err
 	}
+
+	// In case of a directory, we should check its readability in a special way.
+	// In other files, we should not check its Mode explicitly, because the runtime
+	// spec does not mandate the type of masked files. It could be a regular file
+	// or a character file (/dev/null), which is the case for runtimes like runc.
```

Comment:
I prefer [keeping the explicit mode checks][1] to avoid calling `testFileWriteAccess` on types where its logic doesn't make sense.  And if runtimes seek to block write access by binding `/dev/null` over the path in question, we'll need a more sophisticated check than [what `testFileWriteAccess` does now][2], because you can certainly write a byte to `/dev/null`.

[1]: https://github.com/opencontainers/runtime-tools/pull/625#discussion_r182495793
[2]: https://github.com/opencontainers/runtime-tools/blob/9185c46fb7f02a165b0dac8379caab8261b4a74c/cmd/runtimetest/main.go#L459-L464

Reply:
@wking I'm not sure I understand. Can you please elaborate how we can do a more sophisticated check for writeability of `/dev/null`?
I don't see a general way to check if a character device can be written. AFAIK the read/write behavior varies a lot for each character device. The best we can do is to check for the file mode, isn't it?